src/Components/SysTable/SysTable.js

```diff
@@ -1,183 +1,173 @@
-
-    const onSort = (e, index, direction) =>
-        stateSet({ type: 'setSortBy', payload: { index, direction }, tableVars: { orderBy: orderBy({ index, direction }), offset: 0 } });
-
-    const buildFilterChips = () => {
-        const chips = [];
-        const hasMatch = sysTableFilters().condition?.hasMatch !== undefined && (sysTableFilters().condition?.hasMatch ?
-            intl.formatMessage(messages.matched) : intl.formatMessage(messages.notMatched));
-        tableVars?.name &&
-        chips.push({ category: intl.formatMessage(messages.name), value: 'name',
-            chips: [{ name: tableVars?.name, value: tableVars?.name }] });
-        hasMatch && chips.push({ category: intl.formatMessage(messages.status), value: 'matched', chips: [{ name: hasMatch, value: hasMatch }] });
-        return chips;
-    };
-
-    const activeFiltersConfig = {
-        deleteTitle: intl.formatMessage(messages.resetFilters),
-        filters: buildFilterChips(),
-        onDelete: (event, itemsToRemove, isAll) => {
-            if (isAll) {
-                sysTableFilters({});
-                stateSet({ type: 'setTableVars', payload: { name: '' } });
-            } else {
-                itemsToRemove.map((item) => {
-                    item.value === 'name' && stateSet({ type: 'setTableVars', payload: { name: '' } });
-                    item.value === 'matched' && sysTableFilters({});
-                });
-            }
-        }
-    };
-
-    useEffect(() => {
-        const rowBuilder = data => data?.hostsList?.flatMap((data, key) => ({
-            rowId: key,
-            cells: [
-                { title: <Link to={`/systems/${data.displayName}`}>{data.displayName}</Link> },
-                { title: <StatusLabel isDisabled={data.isDisabled} hasMatch={data.hasMatch} displayMatch /> },
-                {
-                    title: data.lastMatchDate ?
-                        <Tooltip content={<DateFormat date={new Date(data.lastMatchDate)} type='exact' />}>
-                            <span><DateFormat date={new Date(data.lastMatchDate)} type='onlyDate' /></span>
-                        </Tooltip>
-                        : <Tooltip content={intl.formatMessage(messages.noHostHas)}>
-                            <span>{intl.formatMessage(messages.never)}</span>
-                        </Tooltip>
-                },
-                { title: <Link to={`/systems/${data.displayName}`}>{data.totalMatches?.toLocaleString()}</Link> },
-                { title: <Tooltip content={<DateFormat date={new Date(data.lastScanDate)} type='exact' />}>
-                    <span><DateFormat date={new Date(data.lastScanDate)} /></span>
-                </Tooltip> }
-
-            ]
-        }));
-
-        stateSet({ type: 'setRows', payload: rowBuilder(sysTableData) });
-    }, [intl, sysTableData]);
+    const fetchSystems = useFetchSystems({
+        query: GET_SYSTEM_TABLE
+    });
```

Comment:
I would try wrapping this in `useCallback` hook, otherwise this will be performed on each render and I think it's enough to create this only on initial render
```suggestion
    const fetchSystems = useCallback(useFetchSystems({
        query: GET_SYSTEM_TABLE
    }));
```